client-js/client/utils.ts

```diff
@@ -0,0 +1,67 @@
+import Bowser from "bowser";
+
+import {
+  name as packageName,
+  version as packageVersion,
+} from "../package.json";
+import { AboutClientData } from "../rtvi/messages";
+
+interface JSAboutClientData extends AboutClientData {
+  platform_details: {
+    browser?: string;
+    browser_version?: string;
+    platform_type?: string;
+    engine?: string;
+    device_memory?: number;
+    language?: string;
+    connection?: {
+      effectiveType?: string;
+      downlink?: number;
+    };
+  };
+}
+
+export function learnAboutClient() {
+  let about: JSAboutClientData = {
+    library: packageName,
+    library_version: packageVersion,
+    platform_details: {},
+  };
+  // This uses legacy browser user agent parsing, which we still fall
+  // back to if the User Agent Hints API is not available
+  let navAgentInfo = null;
+  if (window?.navigator?.userAgent) {
+    try {
+      navAgentInfo = Bowser.parse(window.navigator.userAgent);
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    } catch (_) {
+      // void
+    }
+  }
+
+  if (navAgentInfo?.browser?.name) {
```

Comment:
Not reviewing this part too closely, but I'm assuming it's pretty much the same as what we do in daily-js's logging and telemetry code, which has been thoroughly battle-tested?

Reply:
so yes, battle-tested